src/handlers/mozilla-auth0.js

```diff
@@ -90,16 +90,54 @@ class Handler {
     let profile = await new Promise((resolve, reject) =>
       a0.getUser(req.user.sub, (err, prof) => err ? reject(err) : resolve(prof)));
 
-    if (!profile.email_verified) {
-      debug('profile.email is not verified; ignoring profile');
+    if ('active' in profile && !profile.active) {
+      debug('user is not active; rejecting');
       return;
     }
 
-    let user = new User();
-    user.identity = 'mozilla-auth0/' + profile.email;
+    let user = userFromProfile(profile);
     user.expires = new Date(req.user.exp * 1000);
```

Comment:
Mostly because it requires `req`.  After it is returned from this function, the User object is passed to lots of other functions that get to modify it too, so this isn't an unusual operation (we're pretty far from functional programming here, sorry).